addons/account/models/account_move.py

```diff
@@ -2731,6 +2746,14 @@ def _compute_tax_audit(self):
     # CONSTRAINT METHODS
     # -------------------------------------------------------------------------
 
+    @api.constrains('name')
+    def _auto_label(self):
+        for record in self:
+            if not record.name and record.move_id.type in self.env['account.move'].get_purchase_types(True):
+                record.name = " - ".join([x for x in [record.move_id.commercial_partner_id.display_name, record.move_id.ref] if x]) + " (%s)" % (record.move_id.invoice_date or record.move_id.date)
+            if not record.name and record.move_id.type == 'entry':
+                record.name = record.account_id.name
```

Comment:
gné? A constrains acting like a compute? This task is full of surprises...

Reply:
You don't want the name/label to be a computed field, you also don't want it to have a default value set before starting to edit.

Reply:
It seems strange to me to have the account name inside the line name but... ok